src/accessories/switchAccessory.ts

```diff
@@ -38,62 +23,50 @@ export class SwitchAccessory extends MiHomePlatformAccessory {
     this.service.setCharacteristic(this.platform.Characteristic.Name, accessory.context.device.label);
 
     // each service must implement at-minimum the "required characteristics" for the given service type
-    // see https://github.com/homebridge/HAP-NodeJS/blob/master/src/lib/gen/HomeKit.ts
+    // see https://developers.homebridge.io/#/service/Switch
 
     // register handlers for the On/Off Characteristic
     this.service.getCharacteristic(this.platform.Characteristic.On)
-      .on(CharacteristicEventTypes.SET, this.setOn.bind(this))                // SET - bind to the `setOn` method below
-      .on(CharacteristicEventTypes.GET, this.getOn.bind(this));               // GET - bind to the `getOn` method below
+      .on(CharacteristicEventTypes.SET, this.setOn.bind(this))  // SET - bind to the `setOn` method below
+      .on(CharacteristicEventTypes.GET, this.getOn.bind(this)); // GET - bind to the `getOn` method below
+
+    // Update the state of a Characteristic asynchronously instead
+    // of using the `on('get')` handlers.
+    //
+    // Here we update the state.
+    setInterval(() => {
```

Comment:
Why are we creating a random value and setting it to the brightness characteristic at intervals?

Reply:
This was supposed to update the on state, fixed.

Reply:
Still not sure why we're polling for state? Doesn't Homekit request the state of the device when it needs it?

Reply:
There are use cases for updating the state where HomeKit may not request the current state.
For example; you could have an automation setup so that when `Switch X` is turned (on/off); do `y`. Since the switch can be turned off without using HomeKit (for example; using the MiHome app), polling would allow the new state to be sent back to HomeKit to then trigger that automation.

Reply:
If a switch is turned (on/off) using the MiHome app, HomeKit won't request the new state by default, polling would allow the new state to be sent back to HomeKit.

Reply:
Whilst that does seem reasonable won't the fact that the API doesn't return a reliable value for `power_state` mean that polling could potentially wipe out the "correct" value that Homekit has?

Reply:
The API seems reliable in certain circumstances, I tested it with the MiHome app and using the API yesterday, both seemed reliable. Using the device itself or Alexa doesn't seem to be be reliable. But if the device is updated using the MiHome app, the state HomeKit has would be incorrect (which is why we can poll for it), if the device is updated using the device itself or Alexa, the state HomeKit has would also be incorrect (although polling in this instance still wouldn't work, this seems to be something MiHome need to address themselves).

Reply:
Ok, sounds good. Hopefully when they finally come back re the notifications API we might have a nicer solution for that. Good stuff 👍

Reply:
For sure, this is a temporary solution at best for now.